src/execution/index/art/art.cpp

```diff
@@ -324,8 +332,9 @@ void ART::ConstructAndMerge(IndexLock &lock, PayloadScanner &scanner, Allocator
 		D_ASSERT(logical_types[0] == ordered_chunk.data[0].GetType());
 
 		// generate the keys for the given input
-		vector<unique_ptr<Key>> keys;
-		GenerateKeys(ordered_chunk, keys);
+		keys.clear();
+		arena_allocator.Reset();
```

Comment:
I got rid of the `Clear` by overwriting the existing keys in the vector. We need the `Reset` because the keys no longer own the data. Without it we end up with as much memory being allocated at the end as there are keys. That is not necessary, we only need the memory for the keys in the current chunk and can then reuse that memory for the next chunk of keys.